### PR TITLE
Fix compile options for plugins

### PR DESCRIPTION
**Pull request by Meakk**
Fix coverage, sanitizers and strict build with plugins

### CODE REVIEW

**Review by mwestphal — APPROVED** [review]
Does this have an impact on externally built plugin ?

**Comment by Meakk** [comment]
@mwestphal please review again, I fixed a lot of things.

File: CMakeLists.txt [hunk 1]
```diff
@@ -72,7 +72,7 @@ if(F3D_STRICT_BUILD)
   if(MSVC)
     set(f3d_strict_build_compile_options /W4 /WX)
   else()
-    set(f3d_strict_build_compile_options -Wall -Wextra -Wshadow -Woverloaded-virtual -Wno-deprecated -Wno-strict-overflow -Wno-array-bounds -Wunreachable-code -Wno-missing-field-initializers -Wno-unused-parameter -Wmissing-declarations -Wredundant-decls -Wpointer-arith -Werror)
```
Comment (mwestphal) [inline CMakeLists.txt:75]: why removing it ?

File: plugins/occt/module/vtkF3DOCCTReader.cxx [hunk 2]
```diff
@@ -1,5 +1,10 @@
 #include "vtkF3DOCCTReader.h"
 
+#if defined(__GNUC__)
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Woverloaded-virtual"
```
Comment (mwestphal) [inline plugins/occt/module/vtkF3DOCCTReader.cxx:5]: not fixable ?

**Comment by mwestphal** [comment]
reader.h coverage is lower than expected. Do you know why these methods are not hit ?

```
   * The reader having the highest score (from 0 to 100) is used to read the file.
   * Default is 50.
   */
  virtual int getScore() const { return 50; }

  /**
   * Create the geometry reader (VTK reader) for the given filename
   */
  virtual vtkSmartPointer<vtkAlgorithm> createGeometryReader(const std::string&) const
  {
    return nullptr;
  }
```

File: plugins/occt/module/vtkF3DOCCTReader.cxx (hunk 2 above)
Comment (Meakk) [reply plugins/occt/module/vtkF3DOCCTReader.cxx:5]: No, it's in the OCCT headers

**Comment by Meakk** [comment]
> reader.h coverage is lower than expected. Do you know why these methods are not hit ?
> 
> ```
>    * The reader having the highest score (from 0 to 100) is used to read the file.
>    * Default is 50.
>    */
>   virtual int getScore() const { return 50; }
> 
>   /**
>    * Create the geometry reader (VTK reader) for the given filename
>    */
>   virtual vtkSmartPointer<vtkAlgorithm> createGeometryReader(const std::string&) const
>   {
>     return nullptr;
>   }
> ```

I fixed the `getScore` by overriding it only if the CMake argument is set.  
Regarding `getGeometryReader`, it's a bit more difficult. The base implementation would be called only if the plugin is defining no importer and no reader, which makes no sense.

**Comment by mwestphal** [comment]
> Regarding getGeometryReader, it's a bit more difficult. The base implementation would be called only if the plugin is defining no importer and no reader, which makes no sense.

Then I'm not sure we should provide an implementation then.

**Comment by mwestphal** [comment]
> Then the impl should be {} since it cant be reached.
 
 It is already. great.

**Pull request merged**